src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
+
```

Comment:
use reduce to eliminate  need for a separate loop for summation, use destructuring is good pratice also

```suggestion
  const children = people.filter(
    ({ mother, sex }) =>
      people.some((person) => person.name === mother) &&
      (!onlyWithSon || sex === "m")
  );

  const ageDifferenceSum = children.reduce((sum, { born, mother }) => {
    const motherData = people.find((person) => person.name === mother);
    return sum + born - motherData.born;
  }, 0);
```